test/Microsoft.AspNet.Mvc.Xml.Test/XmlDataContractSerializerInputFormatterTest.cs

```diff
@@ -80,6 +79,24 @@ public void CanRead_ReturnsTrueForAnySupportedContentType(string requestContentT
         }
 
         [Fact]
+        public async Task XmlDataContractSerializer_CachesSerializerForType()
+        {
+            // Arrange
+            var input = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>" +
+                "<DummyClass><SampleInt>10</SampleInt></DummyClass>";
+            var formatter = new TestXmlDataContractSerializerInputFormatter();
+            var contentBytes = Encoding.UTF8.GetBytes(input);
+            var context = GetInputFormatterContext(contentBytes, typeof(DummyClass));
+
+            // Act
+            formatter.CanRead(context);
```

Comment:
I see that you are trying to invoke in such a way that the CreateSerializer is potentially invoked more than once...but this intent is not clear here...can you call `await formatter.ReadAsync(...)` two times instead (by passing in newed up context)...